### ISSUE TITLE
Write service bus HubLifetimeManager

### ISSUE THREAD

**Issue by davidfowl** [aspnet/SignalR#501]

**Comment by muratg** [comment]
Tracking servicebus work here.

**Comment by davidfowl** [comment]
I took a stab at this and the biggest blocker is the fact that the netstandard 2.0 client doesn't support any creation APIs. We can look at using the REST APIs to create the topics and subscriptions but it wouldn't work based solely on the connection string so it's more inconvenient.

**Comment by mpseidel** [comment]
Looking forward to the Service Bus integration. I am looking for a way to host a web app in Azure with reliable scalable collaboration via websockets and I can't find any solution. Socket.io 2 is nice but no Service Bus integration either. 

Could I use the old SignlaR or are there other options that run 100% PaaS on Azure?

**Comment by Misiu** [comment]
@davidfowl what about version [3.0.0-preview-01](https://github.com/Azure/azure-service-bus-dotnet/releases/tag/3.0.0-preview-01)?
If that package doesn't have required APIs maybe we should create Issue for them?
ServiceBus seems to be the easiest way to do scale-out in Azure (at least for me)

**Comment by anurse** [comment]
@mpseidel Is Redis (which has a PaaS on Azure via Azure Redis Cache) unsuitable for your scenario?

@Misiu It's possible that the necessary APIs will have been added. Right now we're well stocked with work items for the 2.1 release, so it's a matter of prioritization. We'll be in contact with the Service Bus team to ensure the APIs we need are present. I'd also be curious if Azure Redis Cache, which SignalR Core fully supports, is suitable for your scenario.

**Comment by mpseidel** [comment]
@anurse I can give Redis Cache a try. Do you know anyone using it in production as a backplan for socket.io or signalr?

**Comment by davidfowl** [comment]
Lots of people are using it as a SignalR backplane on azure today. I don't have numbers but I've spoken to a number of people that use it that way.

We're going to write a backplane for service bus as soon as the APIs are available. I'll take another look and follow up with the team to get an ETA.

**Comment by mpseidel** [comment]
Thanks, I'll give redis-cache with socket.io a try then. 
Looking forward to the new signalr once its ready!

**Comment by davidfowl** [comment]
This is blocked until we get new APIs

**Comment by mjrousos** [comment]
Is this something we expect will be ready for 2.1 RTM or do we not know yet?

**Comment by davidfowl** [comment]
@mjrousos yes, we no longe have any immediate plans to write a service bus backplane. We recommend redis as it's volatile nature is a much better fit for SignalR applications.
